### ISSUE TITLE
and suddenly it doesn't work anymore

### ISSUE THREAD

**Issue by nlsnhrwld** [instaloader/instaloader#1084]
Using istaloader for a while now in different version and no Problem appeared.
Today i tried to download a Hastag and i get this error:

Traceback (most recent call last):
  File "__main__.py", line 464, in <module>
  File "__main__.py", line 461, in __main
  File "__main__.py", line 431, in main
  File "__main__.py", line 149, in _main
  File "instaloader\instaloader.py", line 1023, in download_hashtag
  File "instaloader\structures.py", line 1302, in from_name
  File "instaloader\structures.py", line 1316, in _obtain_metadata
  File "instaloader\structures.py", line 1311, in _query
KeyError: 'graphql'

tried different versions and i figured out that i depends on the login
When i dont login, the downloads start, but after a few you will be redirected to Login page.
-> after a few attempts i cant eben download 1 pic, without being redirected to the login page
Can someone help out whats the matter with the login?
or at least explain whats happening?

using win standalone

**Comment by Showmaster3000** [comment]
you missed important information, at least:
- instaloader version
- command line you used

concerning the limit with redirection to login, see #1067

**Comment by rick-mmi** [comment]
tl;dr hashtag scraping should still work if you're not logged in.

I'm currently using 4.6.2 in production, but I've confirmed that this issue does exist in latest version 4.7.1.

/explore/tags/_?__a=1 endpoint has changed for some users.  Looks like IG is rolling out a change.

The expected return format looks like this: {"graphql":{"hashtag":{"id":"17843783215038189","name":"burrito"

This is the new return format: {"data":{"id":17843783215038189,"name":"burrito"

I only see the new format when logged into instagram.  It looks like the entire structure has changed so Hashtag logic will require a refactor.

**Comment by santinmike** [comment]
I come across  the same error `KeyError: 'graphql'` when running `Hashtag.from_name(L.context, "hashtag_name")`

I am using instaloader 4.7.1

**Comment by Lahmidi-Mustapha** [comment]
i also have the same problem! my code used to work and i didn't change anything.. My version is also 4.7.1

**Comment by sofiavlachou28** [comment]
I also have the same Error (KeyError: 'graphql') when running: `instance.download_hashtag(hashtag="fridakahloinspired",max_count=10)` . My code used to work to. My version: 4.7.1

**Comment by Lahmidi-Mustapha** [comment]
This` Hashtag.from_name(L.context, keyword)` works for me again if i'm not logged in.. Has something with Instaloader or Instagram changed? Does someone know how much requests with `Hashtag.from_name` can i do before i'll be redirected to Instagram Login?

**Comment by Lahmidi-Mustapha** [comment]
@nlsnhrwld Did you find some solutions?

**Comment by nlsnhrwld** [comment]
@Lahmidi-Mustapha  Until now there is no solution, i guess. (My expierence was, that I can download like 6 Pictures until I be redirected)
When i unterstand @rick-mmi  right, there has been a structure change on the Instragram-site, when you are logged in. These changes have to be adapted first in instaloader, if this changes can be adapt. I dont know. But i guess

**Comment by Lahmidi-Mustapha** [comment]
@nlsnhrwld Thanks! i switched to Graph Api from Instagram and it works pretty well and is also stable

**Comment by nlsnhrwld** [comment]
@Lahmidi-Mustapha You need to have a business account for the Instagram API, dont you? and your App also has to be reviewed by facebook, is that right? or is that only if you want to have public pictures?

**Comment by kanarinka** [comment]
I was teaching a workshop with instaloader this morning and 4-5 different people in the workshop encountered this exact issue. I think what @rick-mmi said is the issue - e.g. Instagram has changed its format so instaloader needs to change the Hashtag loading logic.

**Comment by Lahmidi-Mustapha** [comment]
@nlsnhrwld i created an Instagram profile for my Webapp just to retrieve hashtags from there. Then i changed it to a business profile after that i register my personal facebook account as a developper account and from there i could get an Access Token to the Facebook Graph Api. It's works very smooth and fast. I can now make up to 200 request per Hour without getting a too many requests error or putting a sleep() between my requests to avoid them.

**Comment by nyanofthemoon** [comment]
Confirming issue on my end as well in both `4.6.2` and `4.7.1` with python 3.6.

**Comment by jcharvet** [comment]
Having issue too with 4.7.1

**Comment by aandergr** [comment]
Duplicate of #1080.
